backend/src/schema/resolvers/users/location.js

```diff
@@ -110,44 +105,36 @@ const createOrUpdateLocations = async (userId, locationName, driver) => {
   if (data.context) {
     await asyncForEach(data.context, async ctx => {
       await createLocation(session, ctx)
-      try {
-        await session.writeTransaction(transaction => {
-          return transaction.run(
-            `
+      await session.writeTransaction(transaction => {
+        return transaction.run(
+          `
               MATCH (parent:Location {id: $parentId}), (child:Location {id: $childId})
               MERGE (child)<-[:IS_IN]-(parent)
               RETURN child.id, parent.id
             `,
-            {
-              parentId: parent.id,
-              childId: ctx.id,
-            },
-          )
-        })
-        parent = ctx
-      } finally {
-        session.close()
-      }
+          {
+            parentId: parent.id,
+            childId: ctx.id,
+          },
+        )
+      })
+      parent = ctx
     })
   }
   // delete all current locations from user and add new location
-  try {
-    await session.writeTransaction(transaction => {
-      return transaction.run(
-        `
+  await session.writeTransaction(transaction => {
+    return transaction.run(
+      `
           MATCH (user:User {id: $userId})-[relationship:IS_IN]->(location:Location)
           DETACH DELETE relationship
           WITH user
-          MATCH (location:Location {id: $locationId}) 
-          MERGE (user)-[:IS_IN]->(location) 
+          MATCH (location:Location {id: $locationId})
+          MERGE (user)-[:IS_IN]->(location)
           RETURN location.id, user.id
         `,
-        { userId: userId, locationId: data.id },
-      )
-    })
-  } finally {
-    session.close()
-  }
```

Comment:
why are you no longer closing the session here in this function?
is it closed later?

Reply:
See above :point_up: